pkg/storage/test/tuples.go

```diff
@@ -1457,6 +1457,47 @@ func ReadStartingWithUserTest(t *testing.T, datastore storage.OpenFGADatastore)
 		_, objectID := tuple.SplitObject(tuples[0].GetObject())
 		require.Equal(t, "doc1", objectID)
 	})
+	t.Run("enforce_order_of_tuples", func(t *testing.T) {
+		storeID := ulid.Make().String()
+
+		var tupleInReverseOrder = []*openfgav1.TupleKey{
+			tuple.NewTupleKey("document:doc7", "viewer", "user:bob"),
+			tuple.NewTupleKey("document:doc6", "viewer", "user:bob"),
+			tuple.NewTupleKey("document:doc5", "viewer", "user:bob"),
+			tuple.NewTupleKey("document:doc4", "viewer", "user:bob"),
+			tuple.NewTupleKey("document:doc3", "viewer", "user:bob"),
+			tuple.NewTupleKey("document:doc2", "viewer", "group:eng#member"),
+			tuple.NewTupleKey("document:doc1", "viewer", "user:bob"),
+			tuple.NewTupleKey("folder:folder1", "viewer", "user:bob"),
+		}
+
+		err := datastore.Write(ctx, storeID, nil, tupleInReverseOrder)
+		require.NoError(t, err)
+
+		tupleIterator, err := datastore.ReadStartingWithUser(
+			ctx,
+			storeID,
+			storage.ReadStartingWithUserFilter{
+				ObjectType: "document",
+				Relation:   "viewer",
+				UserFilter: []*openfgav1.ObjectRelation{
+					{
+						Object: "user:bob",
+					},
+				},
+			},
```

Comment:
If you add another userfilter, for example `group:fga#member`, what is the expectation around ordering?

Reply:
Should be both object_type and object_id given the tuples yielded otherwise, it will not be ordered when the fields are joined into object within the tupleKey.